### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #159: repo: add additional community docs to the openstack-helm repos
**Kubernetes Version** (output of `kubectl version`): N/A

**Helm Client and Tiller Versions** (output of `helm version`): N/A

**Development or Deployment Environment?**: N/A

**Release Tag or Master**: N/A

**Expected Behavior**: N/A

**What Actually Happened**: N/A

**How to Reproduce the Issue** (as minimally as possible): N/A

**Any Additional Comments**: 
As we prepare our repo for the community, we need to address and include the following in our repository:

- changelog
- code of conduct
- contributing guidelines
- contributor agreements (?)
- expectations (for issues, pr's, contributors and reviewers)
- OWNERS
- RELEASES

### PR DESCRIPTION
**Pull request by wilkers-steve**
Added a CONTRIBUTING.md file to the repo. This gives new contributors
a place to look for details related to contributing to OpenStack-Helm.

**What is the purpose of this pull request?**:  Add a CONTRIBUTING.md file to the project

**What issue does this pull request address?**: Fixes #159 

**Notes for reviewers to consider**:

**Specific reviewers for pull request**: @v1k0d3n

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+## Installation Guide
+
+For those interested in a baremetal deployment, OpenStack-Helm has a [baremetal installation guide](docs/installation/getting-started.md) that walks through a baremetal workflow for getting started.  A baremetal installation gives a better picture of how an OpenStack-Helm deployment might look, but might not be ideal for chart development scenarios.
+
+## Filing Issues
+
+If you have a question about OpenStack-Helm or notice an issue, please feel free to file an issue.  We've provided an issue template for submitting issues, and by providing specific information observed in our charts or a deployment we can address those issues in a timely manner.  If there's not enough information submitted in your issue, we may close the issue and ask you to resubmit your request.
```
Comment (v1k0d3n) [inline CONTRIBUTING.md]: "We've provided an issue template for submitting issues, and by providing specific information observed in our charts or a deployment we can address those issues in a timely manner.  If there's not enough information submitted in your issue, we may close the issue and ask you to resubmit your request."

to 

"Issue templates are leveraged for submitting bugs. By providing specific environmental information, developers can address issues in a timely manner.  If there's not enough detail submitted in the issue, developers may ask for additional information." << I'm just using this opportunity to clean this up.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+1. Make changes related to your issue
+1. Submit a Pull Request, adding WIP tag
+1. Identify errors in TravisCI, if any
+1. Continue to commit required or requested changes
+1. Add required tests and scripts to .travis.yml
+1. When ready, remove WIP tag and request a review
+
+### Development Philosophy
+
+OpenStack-Helm was developed with developers in mind.  We're constantly working to evolve and improve our CI experience to allow developers to get quick, verbose feedback on their changes.  By forking OpenStack-Helm, our developers get our TravisCI file and associated scripts as well.  Any developers should integrate TravisCI with their GitHub repo and test any changes in their fork.  Our developers should also add any required changes to our TravisCI to test and maintain their changes.
```
Comment (v1k0d3n) [inline CONTRIBUTING.md]: We're 

to 

Openstack-Helm developers are constantly...

**Comment by wilkers-steve** [comment]
I'm going to close this issue out until the project moves to a new home, as most of this information assumed we'd be operating under a Github workflow process.  Much of the content here isn't relevant outside of working with Github.  What are your thoughts @v1k0d3n?  We can redraft a contributing doc when we land in a new home.

**Comment by v1k0d3n** [comment]
yup, was thinking the same things @wilkers-steve, but i didn't want to close it without checking with you.

most of the contributing framework, i'd like to pull from [openstack-ansible](https://github.com/openstack/openstack-ansible/blob/master/doc/source/developer-docs/contribute.rst), since those guys have been through the trenches and really care about messaging/docs. if we can make ours look similar in ways, i'd be really happy.

p.s. i'm always going to be stingy about possessive pronouns for potentially high-vis repos/projects, just so users don't feel like there's a push from outside forces other than the collective community.

**Comment by wilkers-steve** [comment]
@v1k0d3n Sounds good.  I agree with the openstack-ansible docs.  

I totally get the wording re: possessive pronouns.

**Pull request closed without merging**